world-client/src/features/chat.js

```diff
@@ -68,7 +68,24 @@ export default class Chat extends Feature {
         li.appendChild(text);
         // Finally add message
         messages.appendChild(li);
+        
+
+        var player = wop.state.getPlayers().find((x) => x.id === message.id);
+        if (!player)
+          break;
+        player.currentMessage = message.text;
+        if (player.chatTimeoutHandle > -1) {
+          clearTimeout(player.chatTimeoutHandle || null);
+        }
+
+        player.setMessage(player.currentMessage)
```

Comment:
You set the `currentMessage` with `setMessage` and then also set it by hand. Settle on one way of setting and displaying a message.